pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,352 @@
+
+			if nc.Status.PodCIDRNAT == liqoconsts.DefaultCIDRValue {
+				remotePodCIDRMsg = NotRemappedMsg
+			} else {
+				remotePodCIDRMsg = nc.Status.PodCIDRNAT
+			}
+			if nc.Status.ExternalCIDRNAT == liqoconsts.DefaultCIDRValue {
+				remoteExternalCIDRMsg = NotRemappedMsg
+			} else {
+				remoteExternalCIDRMsg = nc.Status.ExternalCIDRNAT
+			}
+
+			// Collect Original Network Configs
+			originalSection := selectedSection.AddSection("Original")
+			originalSection.AddEntry("Pod CIDR", nc.Spec.PodCIDR)
+			originalSection.AddEntry("External CIDR", nc.Spec.ExternalCIDR)
+
+			// Collect Remapped Network Configs
+			remoteSection := selectedSection.AddSectionWithDetail("Remapped", remoteSectionMsg)
+			remoteSection.AddEntry("Pod CIDR", remotePodCIDRMsg)
+			remoteSection.AddEntry("External CIDR", remoteExternalCIDRMsg)
+		}
+	}
+	return pic.addVpnSection(ctx, networkSection, foreignCluster.Spec.ClusterIdentity, foreignCluster.Status.TenantNamespace.Local)
+}
+
+// addVpnSection adds a section about the VPN configuration.
+func (pic *PeerInfoChecker) addVpnSection(ctx context.Context, rootSection output.Section,
+	remoteClusterIdentity discoveryv1alpha1.ClusterIdentity, tenantNamespace string) error {
+	te, err := liqogetters.GetTunnelEndpoint(ctx, pic.o.CRClient, &remoteClusterIdentity, tenantNamespace)
+	if err != nil {
+		return err
+	}
+	gwSvcSelector, err := metav1.LabelSelectorAsSelector(&liqolabels.GatewayServiceLabelSelector)
+	if err != nil {
+		return err
+	}
+	gwSvc, err := liqogetters.GetServiceByLabel(ctx, pic.o.CRClient, pic.o.LiqoNamespace, gwSvcSelector)
+	if err != nil {
+		return err
+	}
+	vpnEndpointLocalIP, vpnEndpointLocalPort, err := liqogetters.RetrieveWGEPFromService(
+		gwSvc, liqoconsts.GatewayServiceAnnotationKey, liqoconsts.DriverName)
+	if err != nil {
+		return err
+	}
+	tunnelEndpointSection := rootSection.AddSection("VPN Connection")
+	vpnEndpointSection := tunnelEndpointSection.AddSection("Gateway IPs")
+	vpnEndpointSection.AddEntry("Local", fmt.Sprintf("%s:%s", vpnEndpointLocalIP, vpnEndpointLocalPort))
+	vpnEndpointSection.AddEntry("Remote", fmt.Sprintf("%s:%s",
+		te.Status.Connection.PeerConfiguration[liqoconsts.WgEndpointIP],
+		te.Status.Connection.PeerConfiguration[liqoconsts.ListeningPort],
+	))
+	tunnelEndpointSection.AddEntry("Status", fmt.Sprintf("%s - %s",
+		string(te.Status.Connection.Status),
+		te.Status.Connection.StatusMessage),
+	)
+	tunnelEndpointSection.AddEntry("Latency", te.Status.Connection.Latency.Value)
+	return nil
+}
+
+// addResourceSection adds a section about the resource usage.
+func (pic *PeerInfoChecker) addResourceSection(ctx context.Context, rootSection output.Section,
+	remoteClusterID, localClusterName, remoteClusterName string) error {
+	resInTot, err := resources.GetIncomingTotal(ctx, pic.o.CRClient, remoteClusterID)
+	if err != nil {
+		return err
+	}
+	resOutTot, err := resources.GetOutgoingTotal(ctx, pic.o.CRClient, remoteClusterID)
+	if err != nil {
+		return err
+	}
```

Comment:
You should consider that either, or both, might not exist. This might happen because they are not found (in that case it is fine, no resource is being shared, or for other errors). Either case, the other information should nonetheless be printed.